### DISCUSSION TITLE
Implementation in Kotlin Question

### DISCUSSION THREAD

**Discussion by SmarMykel** [resilience4j/resilience4j#1801] [Q&A] [answered]
Hello,

This might be silly but I am over my head on implementing in kotlin.

When you initiate the retrofit client

`Retrofit retrofit = new Retrofit.Builder()
        .addCallAdapterFactory(CircuitBreakerCallAdapter.of(circuitBreaker))
        .baseUrl("http://localhost:8080/")
        .client(client)
        .build();`

Are you to implement the `CircuitBreakerCallAdapter` yourself or its supposed to be an import. 
Same for the `RetryCallAdapter`

Thanks.

**Comment by RobWin** [comment, marked as answer]
Hi,
support for Retrofit has been dropped.
You can still find the project here: [resilience4j-retrofit](https://github.com/resilience4j/resilience4j-deprecated/tree/master/resilience4j-retrofit)

**Reply by SmarMykel** [reply]
@RobWin Thats unfortunate, so you wouldn't recommend using resilience4j-retrofit in production applications?

**Reply by RobWin** [reply]
You can, but you need to copy the code and maintain it by yourself.
